gearmand/datadog_checks/gearmand/gearmand.py

```diff
@@ -115,6 +115,17 @@ def check(self, instance):
                 message="Connection to %s:%s succeeded." % (host, port),
                 tags=tags,
             )
+            self._collect_metadata(client)
```

Comment:
I think version collected should be in another try/except clause: the service check is about `can.connect` and version collection is a bit out of that scope IMHO. 

I would say, it's same for those two statements above, but, that's probably out of the scope of this PR:

```python
            self._get_aggregate_metrics(tasks, workers, tags)
            self._get_per_task_metrics(tasks, task_filter, tags)
```

Reply:
Addressed in #5927